Networking/Networking/Network/NetworkError.swift

```diff
@@ -41,7 +42,7 @@ extension NetworkError {
         case StatusCode.timeout:
             self = .timeout
         default:
-            self = .unacceptableStatusCode(statusCode: statusCode)
+            self = .unacceptableStatusCode(statusCode: statusCode, response: responseData)
```

Comment:
❓ What do you think about also passing & showing the response in the `notFound` and `timeout` error cases? I'm just a bit worried about missing any possible information like in the DotcomError in support/debugging. Even though "not found" 404 and "timeout" **usually** wouldn't include much meaningful info, it could still contain some response based on the server behavior. Alternatively, we can also consider merging `notFound` and `timeout` cases to `unacceptableStatusCode` - I didn't see any special handling based on these two errors. However, there is a lot of usage in unit tests so this might be a more drastic change.

On the side, I felt like it makes more sense for `invalidURL` and `invalidCookieNonce` cases to be a separate error enum because they aren't related to the error scenarios from the network implementation in `AlamofireNetwork`.

Reply:
> What do you think about also passing & showing the response in the notFound and timeout error cases? I'm just a bit worried about missing any possible information like in the DotcomError in support/debugging.

I agree. 
- Started printing response in https://github.com/woocommerce/woocommerce-ios/pull/11110/commits/082fd6dff4d920450d1e58652ace4e5b5166e5d2
- Added unit tests in https://github.com/woocommerce/woocommerce-ios/pull/11110/commits/b8fd49f11b29dfd0fd3c3cecff12f9a21ec51749
- Used empty paranthesis to initialize the `notFound` and `timeout` error cases in https://github.com/woocommerce/woocommerce-ios/pull/11110/commits/e5e83201e1720cb62d506ef4d5bae76386649c40. This needs to be changed in a lot of files and ends up in 500+ changes. Sorry about that.